### DISCUSSION TITLE
Dialog - Closing by clicking Overlay area suppresses onblur events

### DISCUSSION THREAD

**Discussion by bjeld** [radix-ui/primitives#1689] [Help] [answered]
## Is it intended behaviour that closing a dialog by clicking Overlay area suppresses onblur events?

I have created a sandbox to illustrate my question:
https://codesandbox.io/s/boring-ishizaka-94e4gx?file=/App.js

**Comment by benoitgrelard** [comment, marked as answer]
Hey @bjeld,

Not sure I would say intended behaviour, but rather a consequence of the dismissal on `pointerdown`.
Because `pointerdown` will happen before `blur`.

You can see the same behaviour in this very basic reproduction without Radix:

https://codesandbox.io/s/unmount-on-pointerdown-skips-blur-yspfp7?file=/src/App.js

**Reply by bjeld** [reply]
Thanks for the quick answer.

So guess only solution would be to schedule the execution of the dismissal to run after blur. 
https://codesandbox.io/s/unmount-on-pointerdown-skips-blur-forked-s4p5xm

**Reply by benoitgrelard** [reply]
Yes, but that isn't something we are likely to do internally as it has other side-effects with focus management/animation, etc.

**Reply by bjeld** [reply]
I understand, and Im also not a big fan of adding additional complexity to a codebase. 
I believe our issue can be handled by altering the UX, maybe even for the better. Gonna talk to our produkt team about it.

Anyway, Thank you for the clarification and your time.
